core/src/main/java/io/questdb/cairo/TxReader.java

```diff
@@ -167,17 +167,7 @@ private int findAttachedPartitionIndex(long ts) {
 
     int findAttachedPartitionIndexByLoTimestamp(long ts) {
         // Start from the end, usually it will be last partition searched / appended
-        int hi = attachedPartitions.size() - LONGS_PER_TX_ATTACHED_PARTITION;
-        if (hi > -1) {
-            long last = attachedPartitions.getQuick(hi);
-            if (last < ts) {
-                return -(hi + LONGS_PER_TX_ATTACHED_PARTITION + 1);
-            }
-            if (last == ts) {
-                return hi;
-            }
-        }
-        return attachedPartitions.binarySearchBlock(0, attachedPartitions.size(), LONGS_PER_TX_ATTACHED_PARTITION_MSB, ts);
+        return attachedPartitions.binarySearchBlock(LONGS_PER_TX_ATTACHED_PARTITION_MSB, ts, BinarySearch.SCAN_UP);
```

Comment:
`attachedPartitions` is unique partition timestamp values, adding `SCAN_UP` logic will not make it faster

Reply:
It is a requirement by non-unique algorithm to add scran direction, it's not a matter of performance. The reason I changed the algorithm from unique to non-unique is because I am tripping over it multiple times finding out the hard way the old Algo wasn't working for non-unique values. I feel non-unique algorithm adds very little in terms of overhead but adds quite a lot in terms of saving development time.